app/Http/Controllers/Flights/FlightController.php

```diff
@@ -216,30 +221,29 @@ public function search(Request $request)
      *
      * @param      \Illuminate\Http\Request     $request
      */
-    public function accept(Request $request)
+    public function acceptPublic(Request $request)
     {
         // get the flight to accept
         $flight = Flight::findOrFail($request->id);
 
-        // if the user is the requestee or has already accepted the flight
-        if( $flight->isRequestee(Auth::user()) || $flight->isAcceptee(Auth::user()) )
-        {   // redirect them to the flight
-            return  redirect()
-                    ->route('flights.show', $flight->id)
-                    ->withErrors(['You have already accepted this request!']);
-        }
+        $flight = $this->accept($flight);
 
-        // accept the flight
-        $flight->acceptee_id = Auth::user()->id;
-        $flight->save();
+        // show the flight
+        return redirect()->route('flights.show', ['flight' => $flight]);
+    }
 
-        // send notification
+    /**
+     * Accept a private request
+     * 
+     * @param \Illuminate\Http\Request $request
+     */
+    public function acceptPrivate(Request $request)
+    {
+        $flight = Flight::where('code', $request->code)->first();
 
-        $requestee = $flight->requestee;
-        $requestee->notify(new RequestAccepted(Auth::user(), $flight));
+        $flight = $this->accept($flight);
 
-        // show the flight
-        return redirect()->route('flights.show', ['flight' => Flight::findOrFail($request->id)]);
+        return redirect()->route('flights.show', ['flight' => $flight]);
     }
```

Comment:
Personally, I think this could be better if a single `accept()` method was called, and then an if statement decided which of the `acceptPublic()` or `acceptPrivate()` methods is called. Allows a single route/method to be used for accepting flight requests, but unsure how this would work with the private request code slug.

Reply:
I dont think this would be possible with one method